### ISSUE TITLE
Support for Minecraft 1.12

### ISSUE THREAD

**Issue by EdricChan03** [Chan4077/First-Mod#4]
Is anyone willing to help with this?

- [x] Add advancements
- [x] Upgrade to Forge-1.12 (only when Forge-1.12 comes out of beta)
- [x] Figure out how to initialize `ItemBlock` and `Item`
- [x] Add recipes

**Comment by Chan4077** [comment]
I've managed to do it!

```java
GameRegistry.addRecipe(new ShapedOreRecipe(...))
```
which is to be registered.

**Comment by EdricChan03** [comment]
And it's updated again!
**Shaped Recipe**:
```java
GameRegistry.addShapedRecipe(...)
```
**Shapeless Recipe**:
```java
GameRegistry.addShapelessRecipe(...)
```
(I still need to find out what's passed to `addShapedRecipe` and `addShapelessRecipe`...)

**Comment by EdricChan03** [comment]
This is officially done. Will be closing.

The `1.12.x` source code is now available on the `1.12` branch.
